src/JsonSchema/Validator.php

```diff
@@ -19,37 +20,47 @@
  * @author Bruno Prieto Reis <bruno.p.reis@gmail.com>
  * @see    README.md
  */
-class Validator extends Constraint
+class Validator extends BaseConstraint
 {
     const SCHEMA_MEDIA_TYPE = 'application/schema+json';
 
     /**
      * Validates the given data against the schema and returns an object containing the results
      * Both the php object and the schema are supposed to be a result of a json_decode call.
-     * The validation works as defined by the schema proposal in http://json-schema.org
+     * The validation works as defined by the schema proposal in http://json-schema.org.
+     *
+     * Note that the first argument is passwd by reference, so you must pass in a variable.
      *
      * {@inheritDoc}
      */
-    public function check($value, $schema = null, JsonPointer $path = null, $i = null)
+    public function validate(&$value, $schema = null, $checkMode = null)
     {
+        $initialCheckMode = $this->factory->getConfig();
+        if ($checkMode !== null) {
```

Comment:
so if I understand correctly, `validate` always bulldozes whatever is already  set in the internal checkMode, and, respects the incoming $checkMode, then restores it afterward. If that's true, then shouldn't $checkMode default to CHECK_MODE_NORMAL, and always call `$this->factory->setConfig` whether $checkMode (the arg) is set or not?